src/marten/db/query/sql/query.cr

```diff
@@ -286,20 +286,15 @@ module Marten
             rows_affected.not_nil!
           end
 
-          private def build_count_query
+          private def build_count_query(column_name : String | Symbol | Nil = nil)
             where, parameters = where_clause_and_parameters
             limit = connection.limit_value(@limit)
 
             sql = build_sql do |s|
-              s << "SELECT COUNT(*)"
-              s << "FROM ("
-              s << "SELECT"
-
-              if distinct
-                s << connection.distinct_clause_for(distinct_columns)
-                s << columns
+              if column_name
+                build_count_column_specific_subquery_query(s, column_name)
               else
-                s << "#{Model.db_table}.#{Model.pk_field.db_column!}"
+                build_count_default_subquery_query(s)
```

Comment:
Would it be possible to avoid fowarding the SQL array to the `#build_count_column_specific_subquery_query` and `#build_count_default_subquery_query` methods and simply append the result of these methods to the SQL array instead?

For example:

```crystal
s <<< if column_name
  build_count_column_specific_subquery_query(scolumn_name)
else
  build_count_default_subquery_query
end
```